### PR TITLE
add navitia api calls

### PR DESCRIPTION
**Pull request by nlehuby**
Note that the main of router_tester.py is WIP. It's only used to debug for now.

### CODE REVIEW

File: connectors.py [hunk 1]
```diff
@@ -6,13 +6,59 @@
...
+    origin = "{};{}".format(from_tuple[1], from_tuple[0])
+    destination = "{};{}".format(to_tuple[1], to_tuple[0])
+
+    url_params = {"from" : origin, "to": destination, "first_section_mode[]": fallback_mode, "last_section_mode[]" : fallback_mode}
+    url_params["max_duration"] = 0 #force non PT journey
+    url_params["count"] = 1
+    url_params = dict(url_params, ** additionnal_params)
+
+    url = NAVITIA_URL + "/coverage/{}/journeys".format(coverage)
+    call = requests.get(url, params = url_params,  headers={'Authorization': NAVITIA_TOKEN})
```
Comment (kinnou02) [inline connectors.py]: no space between equals in method call, that's ugly :)

Comment (nlehuby) [inline connectors.py]: fair enough

File: connectors.py [hunk 2]
```diff
@@ -6,13 +6,59 @@
...
+    """
+    Call navitia to compute a non pt journey and extract duration & distance from the appropriate section
+    """
+    if not mode_is_valid(mode):
+        logger.error("Le mode {} est inconnu - valeurs acceptées : walking, biking, driving".format(mode))
+        return
+    mode_mapping = {"walking": "walking", "driving" : "car", "bicycling": "bike"}
+    fallback_mode = mode_mapping[mode]
+    origin = "{};{}".format(from_tuple[1], from_tuple[0])
+    destination = "{};{}".format(to_tuple[1], to_tuple[0])
```
Comment (TeXitoi) [inline connectors.py]: You can mutualize your file parsing with google map.

File: connectors.py [hunk 3]
```diff
@@ -6,13 +6,59 @@
...
+
+    url_params = {"from" : origin, "to": destination, "first_section_mode[]": fallback_mode, "last_section_mode[]" : fallback_mode}
+    url_params["max_duration"] = 0 #force non PT journey
+    url_params["count"] = 1
+    url_params = dict(url_params, ** additionnal_params)
```
Comment (TeXitoi) [inline connectors.py]: I've never seen a space after `**` in this context, but I can't say if that's the recommended way

File: connectors.py [hunk 4]
```diff
@@ -6,13 +6,59 @@
...
+    logger.debug(call.url)
+    if not call.status_code == 200 :
+        logger.error("Appel à navitia KO - status code : {}".format(call.status_code))
+        return
+    navitia_response = call.json()
+    if not "journeys" in navitia_response :
+        logger.error("Pas d'itinéraire retourné par navitia : " + navitia_response["error"]["message"])
+        return
+    navitia_journey = navitia_response['journeys'][0]
+    if not "non_pt" in navitia_journey['tags'] :
```
Comment (TeXitoi) [inline connectors.py]: That'll only work on new_default scenario.

File: router_tester.py [hunk 5]
```diff
@@ -23,10 +23,25 @@ def get_test_cases_from_csv_file(csv_file) :
 if __name__ == '__main__':
 
     for a_test_case in get_test_cases_from_csv_file("./test_cases/auvergne.csv"):
-        test_router = router.get_distance_and_duration_from_google_directions(a_test_case["from"], a_test_case['to'], a_test_case["mode"])
+        print("----- {} ".format(a_test_case))
+        test_router_kraken = router.get_distance_and_duration_from_navitia(a_test_case["from"], a_test_case['to'], a_test_case["mode"])
+        test_router_valhalla = router.get_distance_and_duration_from_navitia(a_test_case["from"], a_test_case['to'], a_test_case["mode"], additionnal_params= {"_override_scenario": "experimental"})
```
Comment (TeXitoi) [inline router_tester.py]: useless space after `additionnal_params=`

File: connectors.py [hunk 6]
```diff
@@ -6,13 +6,59 @@
...
+
+
+def get_distance_and_duration_from_navitia(from_tuple, to_tuple, mode, coverage="fr-auv", additionnal_params={}):
+    """
+    Call navitia to compute a non pt journey and extract duration & distance from the appropriate section
+    """
+    if not mode_is_valid(mode):
+        logger.error("Le mode {} est inconnu - valeurs acceptées : walking, biking, driving".format(mode))
+        return
+    mode_mapping = {"walking": "walking", "driving" : "car", "bicycling": "bike"}
```
Comment (nlehuby) [inline connectors.py]: nope, there is no mode mapping needed with the google api.

File: connectors.py (hunk 2 above)
Comment (nlehuby) [inline connectors.py]: there is no file parsing here ... 
How can I mutualize ? navitia and google do not use the same separator and the lat & lon are not in the same order

Comment (TeXitoi) [inline connectors.py]: My bad, I was thinking that the tuple creation was here.

File: connectors.py (hunk 6 above)
Comment (TeXitoi) [inline connectors.py]: You can do:

```python
mode_mapping = {"walking": "walking", "driving" : "car", "bicycling": "bike"}

def mode_is_valid(mode_candidate):
    return mode_candidate in mode_mapping
```

And use it in google and navitia. That way, you declare only one time the mapping and the set of valid arguments.

**Pull request merged**